### PR TITLE
Add peer heartbeat monitor #36

### PR DESCRIPTION
**Pull request by mkamonMdt**
The commit adds peer heartbeat monitoring functionality that registers each heartbeat received and removes peers that stopped missed all heartbeat opportunities in current monitoring period.

The heartbeat-based peer removal will take up to 2x of monitoring period duration to detect a lost peer.

### CODE REVIEW

**Comment by darioalessandro** [comment]
@mkamonMdt this looks great!!

File: videocall-client/src/client/video_call_client.rs [hunk 1]
```diff
@@ -168,6 +168,24 @@ impl VideoCallClient {
                 })
             },
             on_connection_lost: self.options.on_connection_lost.clone(),
+            peer_monitor: {
+                let inner = Rc::downgrade(&self.inner);
+                let on_connetion_lost = self.options.on_connection_lost.clone();
```
Comment (darioalessandro) [inline videocall-client/src/client/video_call_client.rs:173]: nit: connection

File: videocall-client/src/connection/connection.rs [hunk 2]
```diff
@@ -53,13 +54,17 @@ impl Connection {
...
+        let monitor = options.peer_monitor.clone();
         let mut connection = Self {
             task: Rc::new(Task::connect(webtransport, options)?),
             heartbeat: None,
+            heartbeat_monitor: None,
             status,
             aes,
         };
         connection.start_heartbeat(userid);
+        connection.start_heartbeat_monitor(monitor);
```
Comment (darioalessandro) [inline videocall-client/src/connection/connection.rs:66]: up to you but how about setting `heartbeat_monitor` on 61 as opposed to calling `start_heartbeat_monitor` that way you save the method call to `start_heartbeat_monitor`

**Review by darioalessandro — COMMENTED** [review]
looking good! let me know what you think

File: videocall-client/src/client/video_call_client.rs (hunk 1 above)
Comment (mkamonMdt) [reply videocall-client/src/client/video_call_client.rs]: thx, fixed.

File: videocall-client/src/connection/connection.rs (hunk 2 above)
Comment (mkamonMdt) [reply videocall-client/src/connection/connection.rs]: yeah, that might be better way to do a one-liner, Thx :)

**Review by darioalessandro — APPROVED** [review]

**Comment by darioalessandro** [comment]
Pending testing

**Pull request merged**